monitoring/alerts/spark-rules.yml

```diff
@@ -0,0 +1,98 @@
+      for: 1m
+      labels:
+        severity: major
+      annotations:
+        description: 'Spark dispatcher count unexpectedly decreasing'
+
+    - alert: Spark_Job_Count
+      expr: count(spark_mesos_cluster_executor_count) >= 1000
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of jobs'
+
+    - alert: Spark_Job_Count_Decreasing
+      expr: count(spark_mesos_cluster_executor_count offset 5m) - count(spark_mesos_cluster_executor_count) > 0
+      for: 1m
+      labels:
+        severity: major
+      annotations:
+        description: 'Spark job count unexpectedly decreasing'
+
+    - alert: Spark_CPU_Job
+      expr: count(spark_mesos_cluster_executor_count{task_name=~".*monte.*"}) >= 750
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of CPU jobs'
+
+    - alert: Spark_GPU_Jobs
+      expr: count(spark_mesos_cluster_executor_count{task_name=~".*image.*"}) >= 10
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of GPU jobs'
+
+    - alert: Spark_Streaming_Jobs
+      expr: count(spark_mesos_cluster_executor_count{task_name=~".*Kafka.*"}) >= 250
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of Streaming jobs'
+    
+    - alert: Spark_Dispatcher_Queue_Depth
+      expr: spark_mesos_cluster_driver_waiting > 200
+      for: 3m
+      labels:
+        severity: major
+      annotations:
+        description: 'Spark dispatcher queue depth is beyond desired limits'
+
+    - alert: Spark_Average_Job_Duration
+      expr: (avg(spark_mesos_cluster_driver_launch_to_finish_state_finished_mean) / 60000) >= 75
+      for: 3m
+      labels:
+        severity: major
+      annotations:
+        description: 'Spark jobs taking longer to complete than expected'
+
+    - alert: Spark_Average_Launch_Time
+      expr: (avg(spark_mesos_cluster_executor_start_to_all_launched_mean) / 1000) >= 30
+      for: 3m
+      labels:
+        severity: major
+      annotations:
+        description: 'Spark jobs taking longer to launch than expected'
+
+    - alert: Spark_Launch_Time_Stdev
+      expr: (stddev(spark_mesos_cluster_executor_start_to_all_launched_mean) / 1000) > 20.0
+      for: 3m
+      labels:
+        severity: major
+      annotations:
+        description: 'Spark job launch time has unusually high variance'
+
+    - alert: Spark_Job_Success_Rate
+      expr: (sum(spark_mesos_cluster_driver_finished_count_mesos_state_task_finished) / (sum(spark_mesos_cluster_driver_finished_count_mesos_state_task_finished) + sum(spark_mesos_cluster_driver_finished_count_mesos_state_task_lost) + sum(spark_mesos_cluster_driver_finished_count_mesos_state_task_failed))) >= 99.0
```

Comment:
Is this automatically a percentage? Also, should we not rather (or at least *also*) alert on failure to reach this rate?

Reply:
Hm, how do we know when we fail to reach that value? It could be failing throughout the entirety of the test until the very end. Are you suggesting something similar to what Marcin's doing with reporting the tests in a `steady` state?

Or would you be okay with something a little more simple like
```
alert: Spark_Job_Failure_Rate
   expr: (sum(spark_mesos_cluste... failed))) < 99.0
   for: 1h
```

Reply:
We should be alerting when the rate is lower than the desired threshold. This way one could investigate and do something about it. If it is failing during the entire test then we probably have to rethink how we define this. For example, if no jobs are running, we have a 100% success rate. If one job then starts running, we still have a 100% success rate until it fails, at which point we have a 0% success rate and the alert should trigger.

Reply:
okay i'll change the alert to mark it as a failure if below 99.9%